CONTRIBUTING.md

```diff
@@ -0,0 +1,36 @@
+# Setup
+
+1. In order to contribute you will need to clone/fork the repo and install the requirements locally.
+
+```
+pip install .
+```
+
+2. Install [pre-commit](https://pre-commit.com/).
+
+```
+pip install pre-commit
+pre-commit install
+```
+
+# Workflow
+1. Create and checkout new feature branch for your work.
+
+2. Make changes and add awesome code!
+
+3. Run code-style requirement checks using pre-commit and fix all documentation and typing issues shown in the output.
+
+```
+pre-commit run --all-files -v
```

Comment:
is it possible to run this in the building process of CircleCI?

Reply:
Yes, but I'd suggest we create another workflow in GitHub which checks typing and syntax separately.

Reply:
What is the cons of setting this up in CircleCI as a step before running the tests?

Do you want to use Github actions instead?

Reply:
If there are errors with both styling and tests, it will take multiple runs of CircleCI to identify those.

Yes, using Github actions.

Reply:
> If there are errors with both styling and tests, it will take multiple runs of CircleCI to identify those.

Isn't the case with Github actions too, or does it work in a different way?

I'm fine using Github Actions, at some point we want to migrate all the workflows there. I want to make sure, the styling check runs first and it shouldn't run the tests if something fails.

Reply:
We can have separate workflows for both of them.

Reply:
either way is fine with me, @carlosmondra do you have suggestions?

Reply:
I don't have a strong opinion about this, specially if it's possible to set the code style check in github actions and then run circleci, but I think that's not possible right?

If it's not possible, I'd say that if we are moving toward using Github actions only, maybe we should move the tests to Github actions too instead of adding more stuff to circleci.

Reply:
> I don't have a strong opinion about this, specially if it's possible to set the code style check in github actions and then run circleci, but I think that's not possible right?

this is the part I don't know, is it possible to configure two checks in the PR:
- Github action style check
- CircleCI tests

and we can migrate CircleCI later.

Reply:
It's possible, but I think they are performed at the same time. As far as I know, there is no way to set a github action as a requirement to run circleci, but I may be wrong.

Reply:
I added a workflow using GitHub Actions for style checking. So right now, both runs will run simultaneously and PR cannot be merged until both of them pass.